tests/test_broker.py

```diff
@@ -49,6 +51,58 @@ async def async_magic():
 MagicMock.__await__ = lambda x: async_magic().__await__()
 
 
+def test_lc_decode_listenertype_tcp():
+    """
+    Test we can decode a listener type given as 'tcp'
+    """
+    lc = ListenerConfig("tcp")
+    assert lc.type == ListenerType.TCP
+
+
+def test_lc_decode_listenertype_ws():
+    """
+    Test we can decode a listener type given as 'ws'
+    """
+    lc = ListenerConfig("ws")
+    assert lc.type == ListenerType.WS
+
+
+def test_lc_decode_raw_port():
+    """
+    Test we can decode a raw port number in 'bind'
+    """
+    lc = ListenerConfig(ListenerType.TCP, bind="1883")
```

Comment:
I would assume this is an error. As a bind without a colon would be the ip not adress? I thought.

Reply:
I considered the IP address as being optional but the port number mandatory.  If the address is omitted, it follows the standard Python network library convention of meaning "bind to all interfaces" (like `::` would, but IPv6 *and* IPv4).

I included it as a short-hand for having to write `[::]:1883` or `0.0.0.0:1883`.  `:1883` syntax is kept though as I figured there may be people that still rely on that syntax as a "bind all" in existing configurations.